include/swift/SILOptimizer/Utils/ConstExpr.h

```diff
@@ -82,6 +82,21 @@ class ConstExprEvaluator {
   /// that occur after after folding them.
   void computeConstantValues(ArrayRef<SILValue> values,
                              SmallVectorImpl<SymbolicValue> &results);
+
+  /// Try to decode the specified apply of the _allocateUninitializedArray
+  /// function in the standard library.  This attempts to figure out how the
+  /// resulting elements will be initialized.  This fills in the result with
+  /// a lists of operands used to pass element addresses for initialization,
+  /// and returns false on success.
+  ///
+  /// If arrayInsts is non-null and if decoding succeeds, this function adds
+  /// all of the instructions relevant to the definition of this array into
+  /// the set.  If decoding fails, then the contents of this set is undefined.
+  ///
+  static bool
+  decodeAllocUninitializedArray(ApplyInst *apply, uint64_t numElements,
+                                SmallVectorImpl<Operand*> &elementsAtInit,
+                                SmallPtrSet<SILInstruction*, 8> *arrayInsts);
 };
```

Comment:
@marcrasi  I have a suggestion here. I think this PR could drop all code needed for handling `_allocateUninitializedArray`. If I am not mistaken, it is needed only for initializing array with a non-empty array literal and adds quite a bit of complexity for that. We can later on bring it in, if required. So that piece of code can just be in tensor flow for now. Also, we do plan to add support `array.append` which would make `_allocateUninitializedArray` less essential for populating arrays. What do you think?